core/src/test/java/com/uber/m3/tally/ScopeImplTest.java

```diff
@@ -54,21 +53,21 @@ public void metricCreation() {
 
         Counter sameCounter = scope.counter("new-counter");
         // Should be the same Counter object and not a new instance
-        assertTrue(counter == sameCounter);
+        assertEquals(counter, sameCounter);
 
         Gauge gauge = scope.gauge("new-gauge");
         assertNotNull(gauge);
 
         Gauge sameGauge = scope.gauge("new-gauge");
         // Should be the same Gauge object and not a new instance
-        assertTrue(gauge == sameGauge);
+        assertEquals(gauge, sameGauge);
```

Comment:
This changes the logic of what's happening. You want `assertSame`, not `assertEquals`

Reply:
Good catch!